string.c

```diff
@@ -3154,6 +3154,12 @@ rb_str_substr(VALUE str, long beg, long len)
     return str_substr(str, beg, len, TRUE);
 }
 
+VALUE
+rb_str_substr_with_value_nums(VALUE str, VALUE beg, VALUE len, int empty)
```

Comment:
`value_nums` as in fixnums?

Reply:
Yes?  I originally called it `...with_values`, but then I added the `empty` param which is _not_ a value, so I felt like I needed to the change the name 🤷 
I wish there wasn't already a `rb_str_substr` that took `long`s, this seems like what that function should have been.

Reply:
so the "nums" being the "begin" and "length" args as opposed to the "empty" which is technically a number but used as a bool.

Reply:
Open to suggestions

Reply:
Maybe just `rb_str_substr_two_fixnums` ?

Reply:
I like it